torch/cuda/amp/autocast_mode.py

```diff
@@ -0,0 +1,189 @@
+import torch
+import functools
+import warnings
+import numpy as np
+from torch._six import container_abcs, string_classes
+
+
+class autocast(object):
+    r"""
+    Instances of :class:`autocast` serve as context managers or decorators that
+    allow regions of your script to run in mixed precision.
+
+    In autocast-enabled regions, the backend automatically chooses the precision
```

Comment:
Not "the backend" - that is too vague and mysterious. It is this class that controls the behavior and it should describe what it does.

 You should be clear about what happens when this class is applied. Today it controls the behavior for CUDA operations. Maybe something like "In autocast-enabled regions, CUDA ops cast their inputs to op-specific dtypes before running." We want to be very clear and point users to the list of ops that cast to fp16, the list of ops that cast to fp32, and note that ops not on either list preserve their input dtypes. It should also be clear that although some ops may NOT cast their input dtypes, they may receive DIFFERENT input dtypes by being downstream of other ops. @ngimel points out that "casting inputs" is not an entirely accurate description of the behavior either, as the casting may be internal to the kernel.